### PR TITLE
Bsong/task project nullable/#112

### LINKED ISSUE #112: [Bug] Cannot create task without selecting project
### Steps to reproduce

1. Go to CreateTask component through route `/tasks/new`
2. Fill in all the fields except `Project`
3. Click `ADD`

### Expected results

The task should be added without a linked project

### Actual results

Nothing happens when `ADD` is clicked and the task is not added

### Screenshots

image

### Relevant log output

_No response_

### Severity

Medium

### PR DESCRIPTION
**Pull request by bowsong**
closes #112 

Migrate your databases

### CODE REVIEW

**Review by nasaku898 — APPROVED** [review]
Looks good!

File: backend/src/main/migrations/20220113235621-edit-task-projectId-required.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      onUpdate: 'CASCADE',
+      references: {
+        model: 'project',
+        key: 'id'
+      }
+    });
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.changeColumn('Task', 'projectId', {
```
Comment (Manraj45) [inline backend/src/main/migrations/20220113235621-edit-task-projectId-required.js:18]: why are you doing a "change column" when doing the "down"?

**Review by Manraj45 — COMMENTED** [review]
I had a few question about the code. Can you look at it and let me know?

File: backend/src/main/migrations/20220107212440-add-task-status-enum.js [hunk 2]
```diff
@@ -4,13 +4,17 @@ module.exports = {
...
     ]);
-    
   },
 
   down: async (queryInterface, Sequelize) => {
-    return Promise.all([queryInterface.changeColumn('Task', 'status')]);
-  },
+    return Promise.all([
+      queryInterface.changeColumn('Task', 'status', {
+        type: Sequelize.ENUM('NEW', 'ACTIVE', 'RESOLVED', 'CLOSED', 'REMOVED'),
```
[the remark this answers is not in the document]
Comment (bowsong) [reply backend/src/main/migrations/20220107212440-add-task-status-enum.js:15]: That's for reverting the change when undoing a migration

File: backend/src/main/migrations/20220113235621-edit-task-projectId-required.js (hunk 1 above)
Comment (bowsong) [reply backend/src/main/migrations/20220113235621-edit-task-projectId-required.js:18]: Same thing as the other down

**Review by Manraj45 — APPROVED** [review]
Good job!

**Pull request merged**